server/src/test/java/org/elasticsearch/search/aggregations/bucket/histogram/InternalAutoDateHistogramTests.java

```diff
@@ -77,6 +84,22 @@ protected InternalAutoDateHistogram createTestInstance(String name,
         return new InternalAutoDateHistogram(name, buckets, targetBuckets, bucketInfo, format, pipelineAggregators, metaData);
     }
 
+    public void testGetAppropriateRounding() {
```

Comment:
This test seems to be testing a specific scenario rather than testing the method in general. Could we maybe make the name reflect this so its not assumed that its testing the method in general? Would it also be helpful to add a comment/javaDoc which explains the scenario it is testing?